fastlane_core/lib/fastlane_core/command_executor.rb

```diff
@@ -77,8 +77,13 @@ def execute(command: nil, print_all: false, print_command: true, error: nil, pre
 
         # Exit status for build command, should be 0 if build succeeded
         if status != 0
-          o = output.join("\n")
-          puts(o) unless suppress_output # the user has the right to see the raw output
+          o = String.new
+          is_output_already_printed = print_all && !suppress_output
+          if !is_output_already_printed || !error.nil?
+            o = output.join("\n")
+          end
+          puts(o) unless is_output_already_printed
```

Comment:
Do you also think that this reads a bit easier:
```suggestion
          is_output_already_printed = print_all && !suppress_output
          if is_output_already_printed && error.nil?
            o = ""
          else
            o = output.join("\n")
          end
          puts(o)
```
? 🤔

Reply:
This change breaks behaviour. Will print output twice if is_output_already_printed is false while error block given.

Test `does not print output to stdout when status != 0 and output was already printed` covers this case.

Should add `unless` condition to last row:
```ruby
is_output_already_printed = print_all && !suppress_output
if is_output_already_printed && error.nil?
  o = ""
else
  o = output.join("\n")
end
puts(o) unless is_output_already_printed
```

Reply:
Oh, right, I removed that by mistake. I updated my suggestion. So, does it make sense to you as well?

Reply:
Yeah, will apply it

Should I commit each suggestion separately or collect all edits in 1 new commit?

Reply:
However you feel more comfortable, as we squash merge the PRs